### PR TITLE
wgpu: Implement Context3D.setScissorRectangle

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

**Comment by n0samu** [comment]
Can confirm that this fixes #11161, thank you!

File: render/wgpu/src/context3d/mod.rs [hunk 1]
```diff
@@ -286,6 +290,14 @@ impl WgpuContext3D {
                 .as_ref()
                 .expect("Missing compiled pipeline"),
         );
+        if let Some(rect) = &self.scissor_rectangle {
+            pass.set_scissor_rect(
```
Comment (Dinnerbone) [inline render/wgpu/src/context3d/mod.rs:294]: wgpu will panic if these bounds exceed the render attachment. I don't think we're checking for this, so what does Flash do?

Comment (Aaron1011) [reply render/wgpu/src/context3d/mod.rs:294]: Flash throws an error during `drawTriangles`, but only if `Context3D.enableErrorChecking` is enabled. For now, I think we could just skip setting the scissor rect on the wgpu side, and log a warning.

Comment (Aaron1011) [reply render/wgpu/src/context3d/mod.rs]: I've added in an error log when the rectangle is too large.

File: render/wgpu/src/context3d/mod.rs [hunk 2]
```diff
@@ -286,6 +295,29 @@ impl WgpuContext3D {
                 .as_ref()
                 .expect("Missing compiled pipeline"),
         );
+        if let Some(rect) = &self.scissor_rectangle {
+            let current_size = self.current_texture_size.unwrap();
+            if rect.x_min.to_pixels() as u32 > current_size.width
+                || rect.x_max.to_pixels() as u32 > current_size.width
+                || rect.y_min.to_pixels() as u32 > current_size.height
+                || rect.y_max.to_pixels() as u32 > current_size.height
```
Comment (Lord-McSweeney) [inline render/wgpu/src/context3d/mod.rs:303]: Should this check also happen for negative x/y?

Comment (Aaron1011) [reply render/wgpu/src/context3d/mod.rs:302]: I've adjusted the check

**Review by Dinnerbone — APPROVED** [review]

**Pull request merged**